Tasks/Common/packaging-common/nuget/NuGetToolRunner2.ts

```diff
@@ -46,6 +47,12 @@ function prepareNuGetExeEnvironment(
     let originalCredProviderPath: string = null;
     let envVarCredProviderPathV2: string = null;
 
+    // Set NUGET_PLUGINS_CACHE_PATH to work aroud the NuGet issue with long paths: https://github.com/NuGet/Home/issues/7770
+    var tempDir = tl.getVariable('Agent.TempDirectory');
+    var nugetCacheDir = path.join(tempDir, "NuGetCache");
+    env["NUGET_PLUGINS_CACHE_PATH"] = nugetCacheDir;
```

Comment:
Consider only doing the override if we know it actually shortens the path (comparing to expanded %localappdata%\NuGet\plugins-cache when on windows).
That is depending on Agent.TempDirectory it could, although unlikely, increase the path size.

If it's not easy to do this though, feel free to skip.

Reply:
This is a good idea but skipping this time